M/MuJoCo/bundled/patches/macos.patch

```diff
@@ -0,0 +1,389 @@
+-        check_c_source_compiles("int main() {}" SUPPORTS_DEAD_STRIP)
+-        if(SUPPORTS_DEAD_STRIP)
+-          set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -Wl,-dead_strip)
+-        endif()
+-      endif()
+     endif()
+   endif()
+ 
+diff --git a/cmake/MujocoOptions.cmake b/cmake/MujocoOptions.cmake
+index 998b966..213f422 100644
+--- a/cmake/MujocoOptions.cmake
++++ b/cmake/MujocoOptions.cmake
+@@ -86,7 +86,6 @@ get_mujoco_extra_link_options(EXTRA_LINK_OPTIONS)
+ 
+ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT MSVC))
+   set(EXTRA_COMPILE_OPTIONS
+-      -Werror
+       -Wall
+       -Wimplicit-fallthrough
+       -Wunused
+diff --git a/sample/cmake/MujocoLinkOptions.cmake b/sample/cmake/MujocoLinkOptions.cmake
+index 242767f..f8bb131 100644
+--- a/sample/cmake/MujocoLinkOptions.cmake
++++ b/sample/cmake/MujocoLinkOptions.cmake
+@@ -45,18 +45,6 @@ function(get_mujoco_extra_link_options OUTPUT_VAR)
+           set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=gold)
+         endif()
+       endif()
+-
+-      set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,--gc-sections")
+-      check_c_source_compiles("int main() {}" SUPPORTS_GC_SECTIONS)
+-      if(SUPPORTS_GC_SECTIONS)
+-        set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -Wl,--gc-sections)
+-      else()
+-        set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,-dead_strip")
+-        check_c_source_compiles("int main() {}" SUPPORTS_DEAD_STRIP)
+-        if(SUPPORTS_DEAD_STRIP)
+-          set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -Wl,-dead_strip)
+-        endif()
+-      endif()
+     endif()
+   endif()
+ 
+diff --git a/sample/cmake/SampleOptions.cmake b/sample/cmake/SampleOptions.cmake
+index 998b966..213f422 100644
+--- a/sample/cmake/SampleOptions.cmake
++++ b/sample/cmake/SampleOptions.cmake
+@@ -86,7 +86,6 @@ get_mujoco_extra_link_options(EXTRA_LINK_OPTIONS)
+ 
+ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT MSVC))
+   set(EXTRA_COMPILE_OPTIONS
+-      -Werror
+       -Wall
+       -Wimplicit-fallthrough
+       -Wunused
+diff --git a/simulate/CMakeLists.txt b/simulate/CMakeLists.txt
+index d00c5c3..bba48d7 100644
+--- a/simulate/CMakeLists.txt
++++ b/simulate/CMakeLists.txt
+@@ -36,10 +36,10 @@ project(
+ 
+ enable_language(C)
+ enable_language(CXX)
+-if(APPLE)
+-  enable_language(OBJC)
+-  enable_language(OBJCXX)
+-endif()
++
++enable_language(OBJC)
++enable_language(OBJCXX)
++
+ 
+ option(SIMULATE_BUILD_EXECUTABLE "Build the simulate executable binary." ON)
+ option(SIMULATE_GLFW_DYNAMIC_SYMBOLS "Whether to resolve GLFW symbols dynamically." OFF)
+@@ -136,10 +136,10 @@ target_compile_options(libsimulate PRIVATE ${MUJOCO_SIMULATE_COMPILE_OPTIONS})
+ target_link_libraries(libsimulate PUBLIC lodepng mujoco::platform_ui_adapter mujoco::mujoco)
+ target_link_options(libsimulate PRIVATE ${MUJOCO_SIMULATE_LINK_OPTIONS})
+ 
+-if(APPLE)
```

Comment:
I wouldn't trust it. Internally, CMake has some unset/set calls for `APPLE`, so those might override the variable in certain cases

@JamieMair do you know why it needs to be patched to ignore the check? I thought it should have already been set on the macOS targets.

Reply:
I was having issues when compiling the other targets that has now been set to "off", but I didn't remove them. Now I should have cleaned up the patch files so that the only changes remaining are necessary.